core/state_transition.go

```diff
@@ -273,7 +266,7 @@ func (st *StateTransition) TransitionDb(owner common.Address) (ret []byte, usedG
 		// sufficient balance to make the transfer happen. The first
 		// balance transfer may never fail.
 		if vmerr == vm.ErrInsufficientBalance {
-			return nil, 0, false, vmerr, nil
+			return nil, 0, false, nil, vmerr
```

Comment:
Can you explain this change? It is a sensitive part of the code.

Reply:
I am convinced this will introduce a bug and should be resolved.

Reply:
I agree. I checked history geth commit: https://github.com/ethereum/go-ethereum/commit/b9df7ecdc3d3685180ceb29665bab59e9f614da5
Although the variable name is vmerr, this error means no balance to pay for the tx. It should be consensus err and should be returned in 4-th `err` place.
I'll revert this